### PR TITLE
Prepare for 0.14.0-rc1 release and automate sync process

### PR DESCRIPTION
**Pull request by sivanantha321**
**What this PR does / why we need it**:
- Syncs the helm chart with the latest kustomization configs
- Automates the sync process of generated resources

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Type of changes**
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note

```

**Re-running failed tests**

- `/rerun-all` - rerun all failed workflows.
- `/rerun-workflow <workflow name>` - rerun a specific failed workflow. Only one workflow name can be specified. Multiple /rerun-workflow commands are allowed per comment.

### CODE REVIEW

File: charts/kserve-resources/templates/clusterrole.yaml [hunk 1]
```diff
@@ -307,27 +169,12 @@ rules:
...
+  - clusterservingruntimes/status
+  - inferencegraphs/status
+  - inferenceservices/status
+  - servingruntimes/status
   - trainedmodels/status
   verbs:
   - get
   - patch
   - update
-
```
Comment (yuzisun) [inline charts/kserve-resources/templates/clusterrole.yaml:322]: Can we add a flag to enable/disable this role ? I think currently it is still used with Kafka eventing

File: charts/kserve-resources/templates/clusterrole.yaml [hunk 2]
```diff
@@ -12,50 +11,12 @@ rules:
...
   verbs: ["create"]
-
 ---
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRole
 metadata:
   name: kserve-manager-role
 rules:
-- apiGroups:
-  - admissionregistration.k8s.io
```
Comment (yuzisun) [inline charts/kserve-resources/templates/clusterrole.yaml:23]: are we removing the admission webhook roles ?

File: charts/kserve-crd/templates/serving.kserve.io_clusterlocalmodels.yaml [hunk 3]
```diff
@@ -0,0 +1,86 @@
+---
+apiVersion: apiextensions.k8s.io/v1
```
Comment (yuzisun) [inline charts/kserve-crd/templates/serving.kserve.io_clusterlocalmodels.yaml:1]: might need to rebase from the master, I think this is already there

File: charts/kserve-resources/templates/clusterrole.yaml (hunk 2 above)
Comment (sivanantha321) [reply charts/kserve-resources/templates/clusterrole.yaml:23]: No, It is rearranged https://github.com/kserve/kserve/pull/3970/files#diff-476b4c5d97f5649cb09860e052989f34d1e2c464ba6efd27a98b482cb0741f1eR60

File: charts/kserve-resources/templates/clusterrole.yaml (hunk 1 above)
Comment (sivanantha321) [reply charts/kserve-resources/templates/clusterrole.yaml:322]: Added

File: charts/kserve-crd/templates/serving.kserve.io_clusterlocalmodels.yaml (hunk 3 above)
Comment (sivanantha321) [reply charts/kserve-crd/templates/serving.kserve.io_clusterlocalmodels.yaml:1]: done

File: Makefile [hunk 4]
```diff
@@ -137,19 +137,17 @@ manifests: controller-gen
...
+	cat config/rbac/role.yaml >> charts/kserve-resources/templates/clusterrole.yaml
+	# Copy the local model role with Helm chart while keeping the Helm template condition
+	echo '{{- if .Values.kserve.localmodel.enabled }}' > charts/kserve-resources/templates/localmodel/role.yaml
+	cat config/rbac/localmodel/role.yaml >> charts/kserve-resources/templates/localmodel/role.yaml
+	echo '{{- end }}' >> charts/kserve-resources/templates/localmodel/role.yaml
+	
 	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths=./pkg/apis/serving/v1alpha1
 	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths=./pkg/apis/serving/v1beta1
 
-	#TODO Remove this until new controller-tools is released
```
Comment (yuzisun) [inline Makefile:143]: nice! we can remove this now

File: Makefile [hunk 5]
```diff
@@ -137,19 +137,17 @@ manifests: controller-gen
 	$(CONTROLLER_GEN) $(CRD_OPTIONS) paths=./pkg/apis/serving/... output:crd:dir=config/crd/full
 	$(CONTROLLER_GEN) rbac:roleName=kserve-manager-role paths={./pkg/controller/v1alpha1/inferencegraph,./pkg/controller/v1alpha1/trainedmodel,./pkg/controller/v1beta1/...} output:rbac:artifacts:config=config/rbac
 	$(CONTROLLER_GEN) rbac:roleName=kserve-localmodel-manager-role paths=./pkg/controller/v1alpha1/localmodel output:rbac:artifacts:config=config/rbac/localmodel
+	# Copy the cluster role to the helm chart
```
Comment (greenmoon55) [inline Makefile:140]: noob question, does it have to be in the same file `role.yaml` in helm?

Comment (yuzisun) [reply Makefile:140]: not necessarily but since we are testing primarily using the config directory so it is good to keep the same as helm

**Review by yuzisun — APPROVED** [review]

**Pull request merged**